### DISCUSSION TITLE
Hi, sir. i try to load a custom Material to Model, but it's not show any color.

### DISCUSSION THREAD

**Discussion by amin2312** [jnsmalm/pixi3d#20] [Q&A]
[code]

let light = Object.assign(new PIXI3D.Light, { type: PIXI3D.LightType.directional, x: 0, y: 0, z: 1 });
PIXI3D.LightingEnvironment.main.lights.push(light);

let ball = PIXI3D.Model.from(ResLoader.resources['image_3d/ball.gltf'].gltf,
{
	create: function (source: any): any
	{
		let material = new PIXI3D.StandardMaterial();
		material.normalTexture = PIXI.Texture.from('image_3d/ball.png');
		return material;
	}
});

but the result don't contains any color.

![abc](https://user-images.githubusercontent.com/4578021/109160277-8ab50280-77b0-11eb-96be-e81bb0efa9d4.png)

the raw texture file, as below:
![0](https://user-images.githubusercontent.com/4578021/109160395-b3d59300-77b0-11eb-96ae-d23f25dc8507.png)

**Comment by amin2312** [comment]
the ball.gltf is a standard sphere

**Comment by jnsmalm** [comment]
I you just want to change texture for a material you can do it like this also:

```javascript
let ball = PIXI3D.Model.from(ResLoader.resources['image_3d/ball.gltf'].gltf)
ball.meshes[0].material.normalTexture = PIXI.Texture.from('image_3d/ball.png');
```

**Reply by amin2312** [reply]
May be i know problem,
Is the default camera is Orthographic Camera?
If not, how can i change it to Orthographic Camera?

**Reply by amin2312** [reply]
I have viewed code. The default is look like perspective camera.
I want bo create a billiards game, i need Orthographic Camera.
But i don't know how to create it. please help, thank you

**Reply by amin2312** [reply]
Sir, i try to modify the code of "camera.ts"

```
this._projection = new MatrixComponent(this, 16, data => {
        // Mat4.perspective(this._fieldOfView * PIXI.DEG_TO_RAD, this._aspect || this.renderer.width / this.renderer.height, this._near, this._far, data);
        var WW = _this.renderer.width;var WH = _this.renderer.height;var AR = 0.5; Mat4.ortho(-AR * WW, AR * WW, AR * WH, -AR * WH, _this._near, _this._far, data);
});
```
It seems can work. but the render result has some wrongs when i add a directional light(and the initialized rotation of ball is wrong)
![123](https://user-images.githubusercontent.com/4578021/109270128-9227ea80-7848-11eb-966a-8051242e405b.png)

ps: i use the "pbr.js" example to test.
the code below:
```
let app = new PIXI.Application({
  backgroundColor: 0xdddddd, resizeTo: window, antialias: true
})
document.body.appendChild(app.view)
app.loader.add("skybox.cubemap", "assets/environments/autumn/skybox.cubemap")
app.loader.add("ball.gltf", "assets/models/ball/ball.gltf")
app.loader.load((loader, resources) => {
  let skybox = app.stage.addChild(new PIXI3D.Skybox(resources["skybox.cubemap"].texture))

  let model = app.stage.addChild(PIXI3D.Model.from(resources["ball.gltf"].gltf))
  model.scale.set(50);
  model.position.set(30,0,0);

  model = app.stage.addChild(PIXI3D.Model.from(resources["ball.gltf"].gltf))
  model.scale.set(50);
  model.position.set(-30,0,0);

  var light = Object.assign(new PIXI3D.Light, { type: PIXI3D.LightType.directional, intensity:1, x: 0, y: 0, z: 0 });
  PIXI3D.LightingEnvironment.main.lights.push(light);
})
```
the model and texture file below:
[ball.zip](https://github.com/jnsmalm/pixi3d/files/6048120/ball.zip)

**Reply by jnsmalm** [reply]
Try having a directional light which is not located at 0, 0, 0.

**Reply by jnsmalm** [reply]
You can also try to rotate the light.

`light.rotationQuaternion.setEulerAngles(0, 180, 0)`

**Reply by jnsmalm** [reply]
I have added support for orthographic camera, check develop branch.

**Comment by jnsmalm** [comment]
I have added support for orthographic camera, check develop branch.

**Reply by amin2312** [reply]
thank your help
